File: frontend/src/utils/Request.js
import { backendUrl } from '../components/App'

class Request {

  static get({ url='', id='' }={}) {
    return this.send({
      url,
      id
    })
  }

  static post({ url='', data }) {
    return this.send({
      url,
      method: 'POST',
      data,
    })
  }

  static patch({ url='', id='', data }) {
    return this.send({
      url,
      method: 'PATCH',
      id,
      data,
    })
  }

  static delete({ url='', id='' }) {
    return this.send({
      url,
      method: 'DELETE',
      id,
    })
  }

  static send({ url='', method='GET', id='', data }={}) {
    return new Promise((resolve, reject) => {
      fetch(backendUrl + url + '/' + id, {
        method,
        body: JSON.stringify(data),
        headers: {
          'Content-Type': 'application/json',
        },
      })
      .then(res => {
        if (res.ok) {
          resolve(res.json())
        }
        else {
          reject(res.json())
        }
      })
      .catch(res => reject(res.json()))
    })
  }
}

export default Request

File: device/src/match/core/request.py
# -*- coding: utf-8 -*-

import requests
import time
import json

from config import Settings


class Request:
  @staticmethod
  def get(url, params=None, **kwargs):
    ip = Settings.get_setting_options("remote_host")
    while True:
      try:
        return requests.get('http://' + ip + url, params, **kwargs)
      except:
        time.sleep(5)

  @staticmethod
  def post(url, data=None, json=None, **kwargs):
    ip = Settings.get_setting_options("remote_host")
    while True:
      try:
        return requests.post('http://' + ip + url, data, json, **kwargs)
      except:
        time.sleep(5)

  @staticmethod
  def patch(url, data=None, **kwargs):
    ip = Settings.get_setting_options("remote_host")
    while True:
      try:
        if not hasattr(kwargs, "headers"):
          kwargs["headers"] = {}
        kwargs["headers"]["Content-type"] = "application/json"
        data = json.dumps(data)
        return requests.patch('http://' + ip + url, data, **kwargs)
      except:
        time.sleep(5)
